### PR TITLE
Avoid password check if sspi data is present in login request

### PR DESCRIPTION
**Pull request by Deepesh125**
### Description

In T-SQL world, Authentication mechanism would be chosen by the client as opposed to Postgres where authentication method selection would be controlled through pg_hba.conf file on server side.

For example, TDS client such as sqlcmd would not specify any username in login packet when it wants to use Kerberos/NTLM authentication.

So in order to support Kerberos authentication, This commit avoids consulting pg_hba.conf file if sspi data is present in login request during authentication. 

### Issues Resolved
Task: BABEL-3847
Sub-task: BABEL-3849

Signed-off-by: Dipesh Dhameliya <dddhamel@amazon.com>

### Test Scenarios Covered ###
* **Use case based -**

- Tested these changes manually. We will add more test case in future PRs for AD authentication support. 

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tds/src/backend/tds/tdslogin.c [hunk 1]
```diff
@@ -1759,29 +1755,39 @@ TdsClientAuthentication(Port *port)
 			break;
 		case uaMD5:
 		case uaPassword:
-			/*
-			 * If pg_hba.conf specifies that the entry should be authenticated using
-			 * password and the request doesn't contain a password, we should
-			 * throw an error.
-			 */
-			if (!loginInfo->password)
+			/* if sspiLen > 0 then GSS auth is already done at this point */
```
Comment (kuntalghosh) [inline contrib/babelfishpg_tds/src/backend/tds/tdslogin.c:1758]: How about the following change only?
```
if (loginInfo->sspiLen > 0)
{
	Assert(loginInfo->sspi);

	/* Cleanup sspi data. */
	pfree(loginInfo->sspi);
	loginInfo->sspiLen = 0;
	break;
}

/*
 * If pg_hba.conf specifies that the entry should be authenticated using
 * password and the request doesn't contain a password, we should
 * throw an error.
 */
if (!loginInfo->password)
...

```
The assert is valid as loginInfo->sspi is not null if sspiLen>0. (Check FetchLoginRequest)

File: contrib/babelfishpg_tds/src/backend/tds/tdslogin.c [hunk 2]
```diff
@@ -1769,15 +1776,15 @@ TdsClientAuthentication(Port *port)
 				char		hostinfo[NI_MAXHOST];
 
 				pg_getnameinfo_all(&port->raddr.addr, port->raddr.salen,
-								   hostinfo, sizeof(hostinfo),
```
Comment (kuntalghosh) [inline contrib/babelfishpg_tds/src/backend/tds/tdslogin.c]: Unnecessary changes. Please discard the same, otherwise the git blame becomes difficult.

Comment (Deepesh125) [reply contrib/babelfishpg_tds/src/backend/tds/tdslogin.c:1772]: Done

**Review by kuntalghosh — APPROVED** [review]
